src/common/ua.js

```diff
@@ -1,12 +1,46 @@
-/* global browser */
+import { browser } from './consts';
 
 // UA can be overriden by about:config in FF or devtools in Chrome
 // so we'll test for window.chrome.app which is only defined in Chrome
 // and for browser.runtime.getBrowserInfo in Firefox 51+
-export const isChrome = !!window.chrome?.app;
 
-// eslint-disable-next-line import/no-mutable-exports
-export let isFirefox = !!browser.runtime.getBrowserInfo;
-browser.runtime.getBrowserInfo?.().then((info) => {
-  isFirefox = info.name === 'Firefox' && parseFloat(info.version);
+/** @typedef UA
+ * @property {Boolean} isChrome
+ * @property {Boolean | number} isFirefox - boolean initially, Firefox version number when ready
+ * @property {chrome.runtime.PlatformInfo.arch} arch
+ * @property {chrome.runtime.PlatformInfo.os} os
+ * @property {string} browserName
+ * @property {string} browserVersion
+ */
+const ua = {};
+export default ua;
+
+// using non-enumerable properties that won't be sent to content scripts via GetInjected
+Object.defineProperties(ua, {
+  isChrome: {
+    value: !!window.chrome?.app,
+  },
+  isFirefox: {
+    // will be replaced with the version number in ready()
+    value: !!browser.runtime.getBrowserInfo,
+    configurable: true,
+  },
+  ready: {
+    value: Promise.all([
+      browser.runtime.getPlatformInfo(),
+      browser.runtime.getBrowserInfo?.(),
+    ]).then(([{ os, arch }, { name, version } = {}]) => {
+      Object.assign(ua, {
+        arch,
+        os,
+        browserName: name?.toLowerCase() || 'chrome',
+        browserVersion: version || navigator.userAgent.match(/Chrom\S+?\/(\S+)|$/)[1],
```

Comment:
I think the regexp here is incorrect. It should be

```
/Chrome\/([\d.]+)|$/
```

Reply:
`Chrom` prefix matches a theoretical fork that puts `Chromium` instead of `Chrome`
`\S` matches the future versions which may include letters.
On Chrome it works the same as the regexp you're proposing.

Reply:
Oh I was wrong, I thought it was `\s`.
Not a big deal but I think there is little chance to be another name than `/Chrom(?:e|ium)/`.

Reply:
Yep.